### PR TITLE
ensure si doesn't step over anything

### LINKED ISSUE #196: Should :si not step through wrappers?
`si` is a quite "low level" command so perhaps it should not step through wrappers / kwprep when the next statement is a call?

### PR DESCRIPTION
**Pull request by KristofferC**
Fixes #196

### CODE REVIEW

File: src/commands.jl [hunk 1]
```diff
@@ -439,7 +441,8 @@ function debug_command(@nospecialize(recurse), frame::Frame, cmd::Symbol, rootis
             if isa(ret, BreakpointRef)
                 newframe = leaf(frame)
                 cmd0 == :si && return newframe, ret
-                newframe = maybe_step_through_wrapper!(recurse, newframe)
+                is_si || (newframe = maybe_step_through_wrapper!(recurse, newframe))
+                is_si || maybe_step_through_kwprep!(recurse, newframe, istoplevel)
```
Comment (KristofferC) [inline src/commands.jl]: The reason for this addition is the following case:

```jl
julia> f(x, y) = g(x; y=y);

julia> g(x; y=2) = x + y;

julia> h(x, y) = f(x, y);

julia> frame = JuliaInterpreter.enter_call(h, 2, 1);

julia> frame, pc =  debug_command(frame, :s);

julia> frame
Frame for f(x, y) in Main at REPL[3]:1
  1* 1  1 ─ %1 = (:y,)
  2  1  │   %2 = (Core.apply_type)(NamedTuple, %1)
  3  1  │   %3 = (tuple)(y)
⋮
x = 2
y = 1
```

We want the `:s` to have stepped over the kwprep even after stepping over the wrapper.

**Comment by KristofferC** [comment]
Perhaps we can make a new release after this? I want to release Debugger 0.2.0 as well.

**Comment by KristofferC** [comment]
I'll make a new release then?

**Comment by timholy** [comment]
Yes, much needed.

Have you been running the new Revise? Aside from https://github.com/timholy/Revise.jl/pull/243#issuecomment-474251076 any problems with it? I'm a little nervous that all the extra analysis will be rough on slower machines.

**Comment by KristofferC** [comment]
It seems I have been running Revise master, but I'll swap over to `teh/lowered` now and see if I find anything.

> I'm a little nervous that all the extra analysis will be rough on slower machines.

My machine is pretty much a monster so I wouldn't be able to answer that :P

**Pull request merged**